ansible-ipi-install/roles/installer/tasks/25c_disconnected_registry_create.yml

```diff
@@ -0,0 +1,186 @@
+      delegate_to: localhost
+  
+    - name: append auth to pullsecret
+      shell: |
+        echo '{{ pullsecret }}' | jq -c \
+          '.auths += {{ disconnected_auth }}'
+      register: new_pullsecret
+  
+    - name: set pullsecret with new auth
+      set_fact:
+        pullsecret: "  {{ new_pullsecret.stdout }}"
+  
+    - name: Write pullsecret to registry host
+      copy:
+        content: '{{pullsecret}}'
+        dest: "{{ homedir }}/pullsecret.txt"
+        owner: "{{ ansible_user }}"
+        group: "{{ ansible_user }}"
+    
+    - name: generate ssl certs
+      shell: |
+        openssl req \
+          -newkey rsa:4096 \
+            -nodes \
+            -sha256 \
+            -keyout {{ registry_dir_cert }}/domain.key \
+            -x509 -days 365 \
+            -out {{ registry_dir_cert }}/domain.crt \
+            -subj "/C={{ cert_country }}/ST={{ cert_state }}/L={{ cert_locality }}/O={{ cert_organization }}/OU={{ cert_organizational_unit }}/CN={{ ansible_fqdn }}"
+    
+    - name: Copy cert to pki directory
+      copy:
+        src: "{{ registry_dir_cert }}/domain.crt"
+        dest: /etc/pki/ca-trust/source/anchors/domain.crt
+        remote_src: yes
+        group: "{{ ansible_user }}"
+        owner: "{{ ansible_user }}"
+        mode: 0644
+        force: yes
+        backup: yes
+      become: true
+    
+    
+    - name: Update the CA trust files
+      command: update-ca-trust extract
+      become: true
+    
+    - name: Create container to serve the registry
+      shell: |
+        [[ $( podman ps -a --format '{{podman_format_name | quote }}' --filter name={{ pod_name_registry }} ) ]] \
+          && {
+               podman container stop {{ pod_name_registry }}
+               podman container rm {{ pod_name_registry }}
+             }
+    
+        podman create \
+          --name {{ pod_name_registry }} \
+          -p {{ registry_port_container }}:{{ registry_port }} \
+          -v {{ registry_dir_data }}:/var/lib/registry:z \
+          -v {{ registry_dir_auth }}:/auth:z \
+          -e "REGISTRY_AUTH=htpasswd" \
+          -e "REGISTRY_AUTH_HTPASSWD_REALM=Registry" \
+          -e "REGISTRY_HTTP_SECRET=ALongRandomSecretForRegistry" \
+          -e REGISTRY_AUTH_HTPASSWD_PATH=/auth/htpasswd \
+          -v {{ registry_dir_cert }}:/certs:z \
+          -e REGISTRY_HTTP_TLS_CERTIFICATE=/certs/domain.crt \
+          -e REGISTRY_HTTP_TLS_KEY=/certs/domain.key docker.io/library/registry:2
+      become: true
+    
+    - name: Start container
+      command: podman start {{ pod_name_registry }}
+      become: true
+    
+    - name: Mirror remote registry to local
+      command:
+        oc adm release mirror
+          -a "{{ homedir }}/pullsecret.txt"
+          --from={{ release_image | quote }}
+          --to-release-image={{ local_registry | quote }}/{{ local_repo | quote }}:{{ release_version | quote }}
+          --to={{ local_registry | quote }}/{{ local_repo | quote }}
```

Comment:
Official docs mention:
`To create the installation program that is based on the content that you mirrored, extract it and pin it to the release:`
```bash
 oc adm -a ${LOCAL_SECRET_JSON} release extract --command=openshift-install "${LOCAL_REGISTRY}/${LOCAL_REPOSITORY}:${OCP_RELEASE}"
```

Reply:
modified to extract this installer.
pushed in latest commit.